src/main/java/io/epirus/console/deploy/DeployRunner.java

```diff
@@ -0,0 +1,158 @@
+
+public class DeployRunner {
+    public static final String USAGE = "epirus deploy <network>";
+    private final Path workingDirectory;
+    private final Network network;
+    private final Credentials credentials;
+    private final AccountManager accountManager;
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 1) {
+            new DeployRunner(
+                            Network.valueOf(args[0].toUpperCase()),
+                            new AccountManager(
+                                    CliConfig.getConfig(
+                                            CliConfig.getDefaultEpirusConfigPath().toFile()),
+                                    new OkHttpClient()))
+                    .deploy();
+        } else {
+            Console.exitError(USAGE);
+        }
+    }
+
+    public DeployRunner(Network network, AccountManager accountManager, Path workingDirectory) {
+        this.workingDirectory = workingDirectory;
+        this.network = network;
+        Path walletPath = ProjectUtils.loadProjectWalletFile(workingDirectory).get();
+        Path walletPasswordPath =
+                ProjectUtils.loadProjectPasswordWalletFile(workingDirectory).get();
+        this.credentials = ProjectUtils.createCredentials(walletPath, walletPasswordPath);
+        this.accountManager = accountManager;
+    }
+
+    public DeployRunner(Network network, AccountManager accountManager) {
+        this.workingDirectory = Paths.get(System.getProperty("user.dir"));
+        this.network = network;
+        Path walletPath = ProjectUtils.loadProjectWalletFile(workingDirectory).get();
+        Path walletPasswordPath =
+                ProjectUtils.loadProjectPasswordWalletFile(workingDirectory).get();
+        this.credentials = ProjectUtils.createCredentials(walletPath, walletPasswordPath);
+        this.accountManager = accountManager;
+    }
+
+    public void deploy() throws Exception {
+        AccountUtils.accountInit(accountManager);
+        this.accountManager.checkIfAccountIsConfirmed();
+        fundWallet();
+        waitForBalanceUpdate();
+        uploadSolidityMetadata(network, workingDirectory);
+        runGradle(workingDirectory);
+    }
+
+    private void fundWallet() {
+        int tries = 5;
+        while (tries-- > 0) {
+            try {
+                BigInteger accountBalance = accountManager.getAccountBalance(credentials, network);
+                if (accountBalance.equals(BigInteger.ZERO)) {
+                    WalletFunder.fundWallet(
+                            credentials.getAddress(),
+                            Faucet.valueOf(network.getNetworkName().toUpperCase()),
+                            "alex");
+                }
+                return;
+            } catch (Exception e) {
+
+                try {
+                    Thread.sleep(2000);
+                } catch (InterruptedException ex) {
+                    Thread.currentThread().interrupt();
```

Comment:
Probably not something that will stop this from being merged, but in general it would be nice to check the HTTP response code inside the `fundWallet` method & retry there, rather than wrapping the outer method in a try...catch.